### ISSUE TITLE
Show upcoming expenses

### ISSUE THREAD

**Issue by Tealk** [DennisBauer/RecurringExpenseTracker#370]
**Is your feature request related to a problem? Please describe.** 
It can be challenging to track upcoming expenses for the current month within the Recurring Expense Tracker. Users may find it difficult to anticipate how much more they will need to spend by the end of the month, impacting budgeting and financial planning.

**Describe the solution you'd like**
 I would like to add a new display feature that shows the remaining upcoming expenses for the current month. This feature would aggregate all scheduled recurring expenses for the rest of the month, providing a clear overview of how much is still due to be spent. This will help users make more informed financial decisions throughout the month.

**Describe alternatives you've considered, if any** 
An alternative considered is a calendar view displaying all monthly expenses, though this may be less effective in quickly summarizing upcoming financial commitments for the remainder of the month.

**Additional context**
Adding this feature as a dashboard widget or a dedicated section within the main view could make it easily accessible and ensure users can quickly reference their monthly spending commitments.

**Comment by DennisBauer** [comment]
Thanks for the suggestion. What do you think about adding a similar header area like in the overview to the upcoming tab? I do not think it makes sense to show the exact same for remaining in this week, month and year but maybe just this month.

**Comment by Tealk** [comment]
Yes, you're probably right that only the remaining month makes sense. Yes, that's how I imagined it in the overview.

**Comment by lacyyy** [comment]
I'm looking for a related feature, my problem to solve is: "How much will I have to spend, starting today, up until day X?"

Feature ideas:

- Make the "Upcoming Payments" tab chronologically complete by listing all future occurrences for any single recurring payment (Add a toggle for this?). Currently, only the first future occurrences are listed.
- In this chronological view, show the cumulative sum of expenses for different time points (e.g. in between entries?), starting from the current day.
- Alternatively, add a menu where the user inputs a day and it spits out the sum of expenses up until that day.

Thanks for the app!

**Comment by DennisBauer** [comment]
@lacyyy I really liked your idea of showing all upcoming expenses in a timeline. What do you think about this approach? I've attached a video of how it looks like in the comment:
https://github.com/DennisBauer/RecurringExpenseTracker/pull/494

**Comment by Tealk** [comment]
I like it very much.

**Comment by lacyyy** [comment]
That's great! The per-month totals are simple and easy to understand.
If I wanted to determine my sum of payments up until multiple months into the future, I'd have to sum up the month-totals myself. But that's not a big deal for something done rarely.
